File: Problem022.java
import java.util.Collections;
import java.util.List;

public class Problem022 {
    public static void main(String[] args)  {
        List<String> names = Helper.getItems("names.txt");
        Collections.sort(names);

        int total = 0, n = 1;

        for (String s : names) {
            for (int i = 0; i < s.length(); i++) {
                total += n * (s.charAt(i) - 'A' + 1);
            }

            n++;
        }

        System.out.println(total);
    }
}

File: Problem022.hs
import Helper (split)
import Data.List (sort)
import Data.Char (ord)

main = do content <- readFile "names.txt"
          print $ sum $ zipWith score (sort $ split content) [1..]

score name i = i * sum [ord c - ord 'A' + 1 | c <- name]